File: backend/system_rest_api/system_rest_api/src/main/java/com/didem/system_rest_api/repository/ProductRepository.java
package com.didem.system_rest_api.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.didem.system_rest_api.model.Product;

@Repository
public interface ProductRepository extends JpaRepository<Product, Integer> {

	Optional<Product> findByBarcodeCode(String barcodeCode);

}

File: android/BarcodeApp/app/src/main/java/com/didem/barcodeapp/data/repository/ProductRepository.kt
package com.didem.barcodeapp.data.repository

import com.didem.barcodeapp.data.model.Product
import com.didem.barcodeapp.network.RetrofitClient

class ProductRepository {

    //ürün bulmak için fonksiyon tanımlama
    suspend fun getProductByBarcode(barcodeCode: String, token: String): Result<Product> {
        return try {
            val response = RetrofitClient.apiService.getProductByBarcode(barcodeCode, "Bearer $token")
            if (response.isSuccessful) {
                response.body()?.let {
                    Result.success(it)
                } ?: Result.failure(Exception("Ürün bulunamadı"))
            } else {
                Result.failure(Exception("API hatası: ${response.code()}"))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }
}